File: HomeWorkProjects/homework3/src/main/java/com/gmail/petrusevich/volha/homework3/contacts/sorting/SearchContact.java
package com.gmail.petrusevich.volha.homework3.contacts.sorting;

import com.gmail.petrusevich.volha.homework3.contacts.adapter.ContactListAdapter;

public class SearchContact {

    private SearchContact() {
    }

    public static void searchContact(androidx.appcompat.widget.SearchView searchView, final ContactListAdapter contactListAdapter) {
        searchView.setOnQueryTextListener(new androidx.appcompat.widget.SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                return false;
            }

            @Override
            public boolean onQueryTextChange(String newText) {
                contactListAdapter.getFilter().filter(newText);
                return false;
            }
        });
    }

}

File: HomeWorkProjects/homework5/src/main/java/com/gmail/petrusevich/volha/homework5/database/controller/SearchContact.kt
package com.gmail.petrusevich.volha.homework5.database.controller

import com.gmail.petrusevich.volha.homework5.adapter.ContactListAdapter
import com.gmail.petrusevich.volha.homework5.database.controller.ContactViewModel

class SearchContact {

    fun searchContact(searchView: androidx.appcompat.widget.SearchView, adapter: ContactListAdapter, viewModel: ContactViewModel) {
        searchView.setOnQueryTextListener(object : androidx.appcompat.widget.SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String?): Boolean {
                return false
            }

            override fun onQueryTextChange(newText: String?): Boolean {
                val text = "%$newText%"
                val list = viewModel.getSearchContact(text)
                if (list != null) {
                    adapter.updateListContact(list)
                }
                return false
            }
        })
    }


}